Feature: Add action `app-store-connect get-latest-build-number` (#281)
codemagic-ci-cd/cli-tools: docs/app-store-connect/get-latest-build-number.md

get-latest-build-number
=======================

> **Find the highest build number used for given app considering both TestFlight and App Store submissions**

Reviewer: same as above, can potentially ignore `find -> get`, but extra `the` would be nice

```suggestion
**Get the highest build number used for the given app considering both TestFlight and App Store submissions**
```

### Usage
```bash
app-store-connect get-latest-build-number [-h] [--log-stream STREAM] [--no-color] [--version] [-s] [-v]
    [--log-api-calls]
    [--api-unauthorized-retries UNAUTHORIZED_REQUEST_RETRIES]
    [--api-server-error-retries SERVER_ERROR_RETRIES]
    [--disable-jwt-cache]
    [--json]
    [--issuer-id ISSUER_ID]
    [--key-id KEY_IDENTIFIER]
    [--private-key PRIVATE_KEY]
    [--certificates-dir CERTIFICATES_DIRECTORY]
    [--profiles-dir PROFILES_DIRECTORY]
    [--platform PLATFORM]
    APPLICATION_ID_RESOURCE_ID
```
### Required arguments for action `get-latest-build-number`

##### `APPLICATION_ID_RESOURCE_ID`

Application Apple ID. An automatically generated ID assigned to your app
### Optional arguments for action `get-latest-build-number`

##### `--platform=IOS | MAC_OS | TV_OS`

Apple operating systems
### Optional arguments for command `app-store-connect`

##### `--log-api-calls`

Turn on logging for App Store Connect API HTTP requests
##### `--api-unauthorized-retries, -r=UNAUTHORIZED_REQUEST_RETRIES`

> Specify how many times the App Store Connect API request should be retried in case the called request fails due to an authentication error (401 Unauthorized response from the server). In case of the above authentication error, the request is retried usinga new JSON Web Token as many times until the number of retries is exhausted. If not given, the value will be checked from the environment variable `APP_STORE_CONNECT_API_UNAUTHORIZED_RETRIES`. [Default: 3]

Reviewer: ```suggestion
Specify how many times the App Store Connect API request should be retried in case the called request fails due to an authentication error (401 Unauthorized response from the server). In case of the above authentication error, the request is retried using a new JSON Web Token as many times until the number of retries is exhausted. If not given, the value will be checked from the environment variable `APP_STORE_CONNECT_API_UNAUTHORIZED_RETRIES`. [Default: 3]
```

Author: Won't change in this PR.

##### `--api-server-error-retries=SERVER_ERROR_RETRIES`

Specify how many times the App Store Connect API request should be retried in case the called request fails due to a server error (response with status code 5xx). In case of server error, the request is retried until the number of retries is exhausted. If not given, the value will be checked from the environment variable `APP_STORE_CONNECT_API_SERVER_ERROR_RETRIES`. [Default: 3]
##### `--disable-jwt-cache`

> Turn off caching App Store Connect JSON Web Tokens to disk. By default generated tokens are cached to disk to be reused between separate processes, which can can reduce number of false positive authentication errors from App Store Connect API. If not given, the value will be checked from the environment variable `APP_STORE_CONNECT_DISABLE_JWT_CACHE`.

Reviewer: ```suggestion
Turn off caching App Store Connect JSON Web Tokens to disk. By default, generated tokens are cached to disk to be reused between separate processes, which can reduce the number of false positive authentication errors from App Store Connect API. If not given, the value will be checked from the environment variable `APP_STORE_CONNECT_DISABLE_JWT_CACHE`.
```

Author: Won't fix as part of this PR.

##### `--certificates-dir=CERTIFICATES_DIRECTORY`

Directory where the code signing certificates will be saved. Default:&nbsp;`$HOME/Library/MobileDevice/Certificates`
##### `--profiles-dir=PROFILES_DIRECTORY`

Directory where the provisioning profiles will be saved. Default:&nbsp;`$HOME/Library/MobileDevice/Provisioning Profiles`
### Common options

##### `-h, --help`

> show this help message and exit

Reviewer: ```suggestion
Show this help message and exit
```
